### PR TITLE
Custom sdk support

### LINKED ISSUE #1887: Restore proper support for custom SDK
This is one of our [supported scenarios](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/design.md#supported-scenarios) and right now it is mixed up with enabling instrumentations via, e.g.: `OTEL_DOTNET_AUTO_TRACES_ENABLED`.

In the past we had dedicated env vars for that, it seems that we should return to that. We may need to track/document which env vars will not have an effect when a custom SDK is used. This should be supported for all signals.

The old env vars supporting the scenario were removed in release v0.5.0, see the [release notes](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/v0.5.0).

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Support custom sdk.

Fixes #1887 

## Tests

Included in PR.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

## Todo

- Assert exported metrics
- Update documentation with additional details (e.g client has to subscribe to appropriate sources)

### CODE REVIEW

File: CHANGELOG.md
Comment (pellared) [inline CHANGELOG.md:21]:
```suggestion
- Add support for custom OpenTelemetry .NET SDK setup through `OTEL_DOTNET_AUTO_SETUP_SDK`.
```

File: docs/config.md
Comment (pellared) [inline docs/config.md:36]:
```suggestion
| `OTEL_DOTNET_AUTO_SETUP_SDK`        | Controls whether auto-instrumentation should setup OpenTelemetry .NET SDK at startup.                                                                                                                                       | `true`        |
```

File: src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs
Comment (pellared) [inline src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:147]: I am not sure if we should call this. More in next comment.

File: test/test-applications/integrations/TestApplication.CustomSdk/Program.cs
Comment (pellared) [inline test/test-applications/integrations/TestApplication.CustomSdk/Program.cs:46]: The SDK users are calling `AddHttpClientInstrumentation()`. See: https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.Http.

File: test/test-applications/integrations/TestApplication.CustomSdk/Program.cs
Comment (pellared) [inline test/test-applications/integrations/TestApplication.CustomSdk/Program.cs:42]: I think we should document the activity sources. Otherwise, the user would have problem figuring this out. This can be solved as a separate PR.

File: test/test-applications/integrations/TestApplication.CustomSdk/Program.cs
Comment (pellared) [inline test/test-applications/integrations/TestApplication.CustomSdk/Program.cs:53]: We should document how to use this functionality (as separate PR).

**Review by pellared — COMMENTED** [review]
Two main things:

1. I propose to use "setup" instead of "inject". The SDK uses the wording "configuring the tracer", but I think that the term "configure" is too long.
2. I feel that we should not set the lazy instrumentation when in "manual SDK setup" mode.

BTW this is a very nice PR and feel free to disagree with any of my comments 😉 Happy to see you contributing 🎉

File: CHANGELOG.md
Comment (lachmatt) [reply CHANGELOG.md]: fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2069/commits/7314e8899c9434f53fc74b6a415aa55ad01071cf

File: docs/config.md
Comment (lachmatt) [reply docs/config.md]: fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2069/commits/7314e8899c9434f53fc74b6a415aa55ad01071cf

File: src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs
Comment (lachmatt) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:147]: I understood it to be expected behavior in this scenario when we discussed the requirements with @pjanotti

File: test/test-applications/integrations/TestApplication.CustomSdk/TestApplication.CustomSdk.csproj
Comment (Kielek) [inline test/test-applications/integrations/TestApplication.CustomSdk/TestApplication.CustomSdk.csproj:7]: As discussed privately, please avoid VersionOverride. In main folder there is a *Package.props file where versions can be diefined.

**Review by Kielek — APPROVED** [review]

File: test/test-applications/integrations/TestApplication.CustomSdk/TestApplication.CustomSdk.csproj
Comment (lachmatt) [reply test/test-applications/integrations/TestApplication.CustomSdk/TestApplication.CustomSdk.csproj]: fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2069/commits/d62dbc5eedcab7bca73780e970c8e3a63fe6c020

File: src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs
Comment (pellared) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:147]: @pjanotti can you please double-check. I find it weird.

@lachmatt I am approving as I think this can be addressed in a separate PR.

**Review by pellared — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Configurations/ConfigurationKeys.cs
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/ConfigurationKeys.cs:54]: nit: I like better the text in the `config.md`

File: src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs
Comment (pjanotti) [reply src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs:147]: Per discussion during SIG: Mateusz will check the behavior but barring major issues with that we go as it is and refine it later if needed. Initially the setting to setup the SDK or not should only affect that.

**Review by pjanotti — APPROVED** [review]
LGTM. Per SIG meeting let's wait for the confirmation that dual enabling of same instrumentation is not problematic before merging it.

**Review by rajkumar-rangaraj — APPROVED** [review]
Overall directions in this PR looks good me. Nice work!

**Pull request merged**
